tinygrad/shape/shapetracker.py

```diff
@@ -169,12 +168,14 @@ def real_strides(self, ignore_valid=False) -> Tuple[Optional[int], ...]:
       if tidx in valid_vars and not ignore_valid: ret[i] = None
       elif tidx not in idx_vars: ret[i] = 0
     return tuple(ret)
-  def unit_stride_axes(self, ignore_valid=False) -> List[int]: return [i for i,st in enumerate(self.real_strides(ignore_valid)) if st == 1]
+  
+  def unit_stride_axes(self, ignore_valid=False) -> List[int]: 
+    return [i for i,st in enumerate(self.real_strides(ignore_valid)) if st == 1]
 
   def _expr_idx(self, idx, valid):
     for v in reversed(self.views[0:-1]):
-      valid = v.expr_node_mask(idx, valid)
```

Comment:
Is there a reason these can't stay class methods?

Reply:
They could. Adds an extra line per function. In that case the class acts as a namespace - a new file (view.py ?) might be cleaner.

There is quite a lot to consolidate here, once the ShapeTracker class is gone most of its methods are 'View' methods (take in a view, spit out a view). But that is for a next MR.

Reply:
I don't care about line count!

Reply:
There actually is an intricacy here: in Python 3.8, a `staticmethod` is not callable yet as a regular function __from inside the class itself__. 

[This changed in 3.10](https://docs.python.org/3/whatsnew/3.10.html#other-language-changes).

So something like this fails the tests based on Python 3.8:

```
  @staticmethod
  @functools.lru_cache(maxsize=None)
  def expr_idxs(shape:Tuple[int, ...], strides: Tuple[int, ...], offset:int, idxs):
    assert len(idxs) == len(shape), f"need an idx for all dimensions {idxs} vs {shape}"
    nodes = [idx*st for idx,sh,st in zip(idxs, shape, strides) if sh != 1 and st != 0] + ([Node.num(offset)] if offset else [])
    return Node.sum(nodes) if len(nodes) > 1 else nodes[0] if len(nodes) == 1 else Node.num(0)
```
and then 
```
... View.expr_idxs(abc) ... from another function inside the class.
```

There's a way around that: https://discuss.python.org/t/call-staticmethod-from-class-body/14542
... but mypy does not understand that.

However, as far as I understand, the LRU cache of a staticmethod is shared among instances of the class, so calling the function on an instance of the class (e.g. views[-1].expr_node_mask() instead of `View.expr_node_mask()) should work and stay cached.